src/pages/en/guides/baas/appwriteio.mdx

```diff
@@ -0,0 +1,427 @@
+:::
+
+```ini title=".env"
+APPWRITE_ENDPOINT=YOUR_ENDPOINT
+APPWRITE_PROJECT_ID=YOUR_PROJECT_ID
+APPWRITE_DB_ID=YOUR_DATABASE_ID
+APPWRITE_COLLECTION_ID=YOUR_COLLECTION_ID
+...
+```
+
+Now, you should be able to use these environment variables in your project. 
+
+If you would like to have IntelliSense for your Contentful environment variables, you can create a `env.d.ts` file in the `src/` directory and configure `ImportMetaEnv` like this:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly APPWRITE_ENDPOINT: string;
+  readonly APPWRITE_PROJECT_ID: string;
+  readonly APPWRITE_APIKEY: string;
+  readonly APPWRITE_DB_ID: string;
+  readonly APPWRITE_COLLECTION_ID: string;
+}
+```
+:::tip
+Read more about [using environment variables](/en/guides/environment-variables/) and `.env` files in Astro.
+:::
+
+Your root directory should now include these new files:
+
+<FileTree title="Project Structure">
+- src/
+  - **env.d.ts**
+- **.env**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+### Installing dependencies
+
+To connect with your Appwrite project, install both of the following using the single command below for your preferred package manager:
+- [`appwrite`](https://github.com/appwrite/sdk-for-web), the official Appwrite SDK for JavaScript.
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install appwrite
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm install appwrite
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add appwrite
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Next, create a new file called `appwrite.ts` in `src/lib/` directory in your project.
+
+```ts title="src/lib/appwrite.ts"
+import { Client, Databases } from 'appwrite';
+
+export interface BlogPost {
+    title: string;
+    date: string;
+    description: string;
+    content: Document;
+    slug: string;
+}
+
+export const appwriteClient = new Client()
+    .setEndpoint(import.meta.env.APPWRITE_ENDPOINT) // Your API Endpoint
+    .setProject(import.meta.env.APPWRITE_PROJECT_ID); // Your project ID
+```
+
+The above code snippet creates a new Appwrite client, passing in credentials from the `.env` file. 
+
+Finally, your root directory should now include these new files:
+
+<FileTree title="Project Structure">
+- src/
+  - env.d.ts
+  - lib/
+    - **appwrite.ts**
+- .env
+- astro.config.mjs
+- package.json
+</FileTree>
+
+### Fetching data
+
+Include databases in your appwrite.ts file so that you can use this wherever you might need data.
+
+```ts
+export const appwriteDatabases = new Databases(appwriteClient);
+```
+
+Now, Astro components can fetch your data using the `appwriteDatabases`.
+
+For example, if you want to build blog pages.
+
+```astro
+---
+import { appwriteDatabases } from "../lib/appwrite";
+
+interface BlogPost {
+    title: string,
+    content: string
+}
+
+const entries = await databases.createDocument('[DATABASE_ID]', '[COLLECTION_ID]', '[DOCUMENT_ID]', {});
```

Comment:
Won't this just give us empty entries? Shouldn't this be `listDocuments`?

Reply:
This was just an example of one type of use. Maybe it is misleading though and we should switch it to match the rest of the example more closely. We can keep chatting in the discord channel as well :D